### PR TITLE
[INLONG-5117][SortStandalone] Support specify component type from remote config

### LINKED ISSUE #5117: [Feature][Sort-Standalone] Support specify component type from remote config
### Description

image
Current config loading strategy restricts the flexibility of sort-standalone by naming the only type of components in config file.
Which means one sort-standalone node can only support a series of tasks with the same type of sink. 

By specify this params into **_SortTask_**,  the sort-standalone will be decoupled from node to task.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
- Fixes #5117

### Motivation

Current config loading strategy restricts the flexibility of sort-standalone by naming the only type of components in config file. Which means one sort-standalone node can only support a series of tasks with the same type of sink.
By specify this params into SortTask, the sort-standalone will be decoupled from node to task.

### Modifications

allow sort-standalone get type of components from remote config

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/utils/FlumeConfigGenerator.java [hunk 1]
```diff
@@ -32,17 +33,19 @@ public class FlumeConfigGenerator {
     public static final String KEY_SORT_CHANNEL_TYPE = "sortChannel.type";
     public static final String KEY_SORT_SINK_TYPE = "sortSink.type";
     public static final String KEY_SORT_SOURCE_TYPE = "sortSource.type";
+    public static final String KEY_SDK_START_TIME = "start-time";
```
Comment (EMsnap) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/utils/FlumeConfigGenerator.java:36]: CamelCase please

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/utils/FlumeConfigGenerator.java [hunk 2]
```diff
@@ -51,14 +54,15 @@ public static Map<String, String> generateFlumeConfiguration(SortTaskConfig task
...
-    private static void appendChannels(Map<String, String> flumeConf, String name) {
+    private static void appendChannels(Map<String, String> flumeConf, String name, Map<String, String> sinkParams) {
         StringBuilder builder = new StringBuilder();
         String channelName = name + "Channel";
         flumeConf.put(name + ".channels", channelName);
         String prefix = builder.append(name).append(".channels.").append(channelName).append(".").toString();
         builder.setLength(0);
         String channelType = builder.append(prefix).append("type").toString();
-        String channelClass = CommonPropertiesHolder.getString(KEY_SORT_CHANNEL_TYPE);
+        String channelClass = sinkParams.getOrDefault(KEY_SORT_CHANNEL_TYPE,
```
Comment (EMsnap) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/utils/FlumeConfigGenerator.java:64]: final a variable inside a method

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/utils/FlumeConfigGenerator.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/utils/FlumeConfigGenerator.java]: fixed, thx

**Review by healchow — APPROVED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**